src/test.rs

```diff
@@ -362,34 +359,116 @@ fn it_works_with_lots_of_things_codec() {
 
 #[test]
 fn it_works_with_validation() {
-    let method = Method {
-        name: "method_test".to_string(),
-        block_cooldown: Some(123),
-        constraints: None,
-    };
+    {
+        let contract = Contract {
+            data_table: DataTable::new(vec![
+                PactType::Numeric(Numeric(111)),
+                PactType::Numeric(Numeric(333)),
+                PactType::StringLike(StringLike(b"testing")),
+            ]),
+            bytecode: [OpCode::EQ.into(), 0, 0, 1, 0, OpCode::EQ.into(), 0, 1, 1, 1].to_vec(),
+        };
+        let mut constraints: Vec<u8> = Vec::new();
+        contract.encode(&mut constraints);
+
+        let method = Method {
+            name: "method_test".to_string(),
+            block_cooldown: Some(123),
+            constraints: Some(constraints.clone()),
+        };
+
+        let mut methods: Vec<(String, Method)> = Default::default();
+        methods.push((method.name.clone(), method.clone()));
+
+        let module = Module {
```

Comment:
Seems like time to write some helper functions for building `Module` and `Method` structs

Reply:
Can open an issue for this if needed

Reply:
https://github.com/cennznet/cennznut-rs/issues/21#issue-510400919